mllib/src/main/scala/org/apache/spark/mllib/fpm/PrefixSpan.scala

```diff
@@ -44,27 +46,43 @@ class PrefixSpan private (
     private var maxPatternLength: Int) extends Logging with Serializable {
 
   /**
+   * The maximum number of items allowed in a projected database before local processing. If a
+   * projected database exceeds this size, another iteration of distributed PrefixSpan is run.
+   */
+  private val maxLocalProjDBSize: Long = 10000
```

Comment:
Please leave a TODO to make it configurable with a better default value. `10000` may be too small.